### PR TITLE
feat: support alpha values, hsl(), named colors

### PR DESCRIPTION
**Pull request by sgtcoolguy**
The series of commits gives details on each but this PR adds support for:
- named colors
- using alpha values in hex/`rgb()`
- using `hsl()`
It records the input type of color (hex, named, hsl, rgb) and attempts to maintain that type of value when a color is selected and re-inserted (if possible, i.e. named colors will almost always become hex values, but if there's an exact match for the chosen hex it will prefer the named color)

Alpha is basically just ignored and re-inserted (so a color with alpha retains the alpha value).

### CODE REVIEW

**Comment by sergeichestakov** [comment]
awesome, thanks for the PR and apologies for the delay! @masad-frost do you mind taking a look when you get the chance?

File: src/index.ts [hunk 1]
```diff
@@ -20,86 +29,271 @@ function colorPickersDecorations(view: EditorView) {
...
+  }
+
+  return [color, ''];
 }
 
-function rgbComponentToHex(component: number) {
-  const hex = component.toString(16);
+function rgbComponentToHex(component: string): string {
+  let numericValue: number;
+  if (component.endsWith('%')) { // 0-100%
```
Comment (masad-frost) [inline src/index.ts:121]: TIL you can use % for `rgb(`

File: src/index.ts [hunk 2]
```diff
@@ -20,86 +29,271 @@ function colorPickersDecorations(view: EditorView) {
...
+    const percent = Number(component.slice(0, -1));
+    numericValue = Math.round((percent / 100) * 255.0);
+  } else {
+    numericValue = Number(component); // assume 0-255
+  }
+  return decimalToHex(numericValue);
```
Comment (masad-frost) [inline src/index.ts:127]: Wonder if we should handle `NaN` before this

File: src/index.ts [hunk 3]
```diff
@@ -8,8 +8,17 @@ import {
...
-const rgbCallExpRegex = /rgb\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*\)/;
+enum ColorType {
+  rgb = "RGB",
+  hex = "HEX",
+  named = "NAMED",
+  hsl = "HSL"
+};
+
+const rgbCallExpRegex = /rgb\(\s*(\d{1,3}%?)\s*,\s*(\d{1,3}%?)\s*,\s*(\d{1,3}%?)\s*(,\s*0?\.\d+)?\)/;
+const hslCallExpRegex = /hsl\(\s*(\d{1,3})\s*,\s*(\d{1,3})%\s*,\s*(\d{1,3})%\s*(,\s*0?\.\d+)?\)/;
```
Comment (masad-frost) [inline src/index.ts:21]: I was reading the up on HSL and came across the fact that `hsl` and `rgb` in css support space separators instead of commas too!

**Review by masad-frost — APPROVED** [review]
Excellent.

**Comment by masad-frost** [comment]
Gonna merge and bump, feel free to send more updates.

Sorry for keeping this hanging, was away for a while.

File: src/index.ts (hunk 3 above)
Comment (sgtcoolguy) [reply src/index.ts:21]: Yeah I didn't even try and handle the new syntax that supports spaces in places of commas - and also uses a '/' to separate the alpha. I figured the regex would get pretty funky if we wanted to really insist that if you start using command you must use commas for the rest (i.e. not just pop a '?' optional marker on each comma)

**Pull request merged**